### PR TITLE
Implemented upcoming menus and final bug fixes.

### PR DESCRIPTION
**Pull request by antoinztte**
## Overview

Implemented new upcoming menus page, with three kinds of filtering (day selection, menu choice, and campus location).

## Changes Made

1. Created new upcoming menus page and added to tab bar.
2. Created MenusScreen group, comprised of:
  --> MenuCardTableViewCell: contains the EateryExpandableCardDetailView() and EateryExpandableCardContentView(), responsible for each eatery's card in MenusViewController.
  --> MenuDayPickerTableViewCell: responsible for the dayPicker cell in MenusViewController.
  --> MenusFilterViewController: contains five filter pill buttons (i.e. MealType, all, north, west, central) to filter eateries based on campus location. Delegate of UpcomingMenuPickerSheetViewController, and delegator to MenusModelController.
  --> MenusViewController: comprised of a table view that takes in multiple types of cells (i.e. dayPicker, customView, titleLabel, loadingLabel, expandableCard, loadingCard) that makes up the upcoming menu's page. 
 3. Created UpcomingMenuPickerSheetViewController and UpcomingMenuChoiceView, which control the half-sheet displaying Breakfast, Lunch, Dinner, and Late Dinner options.
 4. Created EateryExpandableCardContentView, responsible for showing the eatery name, eatery label, and eatery status of each eatery within the cell (ex. Closing Soon, Opening Soon, etc.). Also includes "Eatery Details" button which displays EateryModelVC for chosen eatery (only for those that are closed or opening soon).
 5. Created EateryExpandableCardDetailView, responsible for showing the menu categories of each eatery based on two filters: day and selected meal type.
 6. Changed downward pointing arrow to new calendar logo within each eatery's page.
 7. Addressed some merge conflicts between my branch and Jason's branch.
 8. Removed wait times.

## Test Coverage

- iPhone XR, iPhone 15 Pro Max, iPhone 15 Pro

## Next Steps 

- Already added eatery hours to each eatery, just can't test until Monday morning. Remind me to commit if test goes well.
- (Unrelated to Upcoming Menus) logging out thinks you're still logged in until you restart the app. Slight bugs with logging in. 

## Screenshots 

Simulator Screenshot - iPhone 15 - 2023-10-09 at 02 17 44

### CODE REVIEW

**Review by vinnie4k — APPROVED** [review]
Great work! Briefly skimmed for styling issues and tested it. There are some minor bugs/UI issues but should be good enough for MVP.

File: Eatery Blue/UI/EateryCard/EateryExpandableCardMenuCategoryView.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        addSubview(categoryNameLabel)
+        
+        categoryNameLabel.snp.makeConstraints { make in
+            make.leading.top.trailing.equalToSuperview()
+        }
+    }
+    
+    private func setupFoodItemLabel() {
+        foodItemLabel.textColor = UIColor.Eatery.gray05
+        foodItemLabel.font = UIFont.preferredFont(for: .footnote, weight: .regular)
```
Comment (tiffany-pann) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardMenuCategoryView.swift:59]: Same here

File: Eatery Blue/UI/EateryCard/EateryExpandableCardMenuCategoryView.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        
+        foodItemLabel.text = String(foodItemString.dropLast(1))
+        categoryNameLabel.text = menuCategory.category
+    }
+    
+    // MARK: - Set Up Views
+    
+    private func setupCategoryNameLabel() {
+        categoryNameLabel.textColor = UIColor.Eatery.black
+        categoryNameLabel.font = UIFont.preferredFont(for: .title3, weight: .semibold)
```
Comment (tiffany-pann) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardMenuCategoryView.swift:48]: You used just `.preferredFont` in the following files and I checked other parts of the codebase and I think it uses just the static var so I'd probably change this to `.preferredFont` for consistency.

File: Eatery Blue/UI/MenuPickerSheet/UpcomingMenuPickerSheetViewController.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    
+    private var menuChoiceViews: [UpcomingMenuChoiceView] = []
+    var selectedMenuIndex: Int?
+    
+    func setUp() {
+        addHeader(title: "Menus")
+        addMenuChoiceViews()
+        addPillButton(title: "Show menu", style: .prominent) { [self] in
+            if let selectedIndex = self.selectedMenuIndex {
+                delegate?.upcomingMenuPickerSheetViewController(self, didSelectMenuChoiceAt: selectedIndex)
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenuPickerSheet/UpcomingMenuPickerSheetViewController.swift:33]: Not sure if you managed to get to this but on call you mentioned that you may only need one of two (either the selectedIndex or the actual menu choice) but if so, you should remove the other one

File: Eatery Blue/UI/MenusScreen/MenusFilterViewController.swift [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+extension MenusFilterViewController: UpcomingMenuPickerSheetViewControllerDelegate {
+    
+    func upcomingMenuPickerSheetViewController(_ vc: UpcomingMenuPickerSheetViewController, didChangeMenuChoice string: String) {
+        mealType.label.text = string
+        if let mealType = mealType.label.text {
+            delegate?.menusFilterViewController(self, didChangeMenuType: mealType)
+        }
+    }
+    
+    func upcomingMenuPickerSheetViewController(_ viewController: UpcomingMenuPickerSheetViewController, didSelectMenuChoiceAt index: Int) { }
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusFilterViewController.swift:161]: Remove if unused

File: Eatery Blue/UI/MenusScreen/MenusFilterViewController.swift [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        all.setHighlighted(filter.north && filter.west && filter.central)
+        north.setHighlighted(filter.north)
+        west.setHighlighted(filter.west)
+        central.setHighlighted(filter.central)
+        
+        func setFilter(_ filter: EateryFilter, animated: Bool) {
+            self.filter = filter
+            updateFilterButtonsFromState(animated: animated)
+        }
+    }
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusFilterViewController.swift:149]: Line break here

File: Eatery Blue/UI/MenusScreen/MenusModelController.swift [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        navigationController?.hero.isEnabled = false
+        navigationController?.pushViewController(viewController, animated: true)
+    }
+    
+}
+
+extension MenusModelController: MenusFilterViewControllerDelegate {
+    
+    func menusFilterViewController(_ viewController: MenusFilterViewController, didChangeMenuType string: String) {
+        currentMealType = string
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusModelController.swift:155]: I probably wouldn't name the param `string` here but not a huge deal

File: Eatery Blue/UI/MenusScreen/MenusNavigationView.swift [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        self.fadeInProgress = progress
+
+        if animated {
+            UIView.animate(withDuration: 0.125) {
+                self.setFadeInProgress(progress)
+            }
+        } else {
+            setFadeInProgress(progress)
+        }
+    }
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusNavigationView.swift:79]: Line break here

File: Eatery Blue/UI/MenusScreen/MenusViewController.swift [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+
+import Combine
+import CoreLocation
+import EateryModel
+import Kingfisher
+import UIKit
+
+/// Used for Expanding Cell
+struct ExpandedEatery {
+    let eatery: Eatery
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusViewController.swift:16]: Line break after struct declaration

File: Eatery Blue/UI/MenusScreen/MenusViewController.swift [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+    var isExpanded: Bool = false
+    var selectedMealType: String?
+}
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusViewController.swift:19]: Line break before the ending }

File: Eatery Blue/UI/MenusScreen/MenusViewController.swift [hunk 10]
```diff
@@ -0,0 +1,350 @@
...
+            return cell
+        }
+    }
+    
+}
+    
+extension MenusViewController: UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section > 2 {
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusViewController.swift:218]: `return section > 2 ? 12 : 0`

File: Eatery Blue/UI/MenusScreen/MenusViewController.swift [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+            return 12
+        }
+        return 0
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if indexPath.section == 0 {
```
Comment (tiffany-pann) [inline Eatery Blue/UI/MenusScreen/MenusViewController.swift:225]: `return indexPath.section == 0 ? 84 : UITableView.automaticDimension`

File: Eatery Blue/UI/PillButton/PillButtonView.swift [hunk 12]
```diff
@@ -53,7 +53,7 @@ class PillButtonView: UIView {
         }
 
         imageView.snp.makeConstraints { make in
-            make.width.height.equalTo(24)
+            make.width.height.equalTo(16)
```
Comment (tiffany-pann) [inline Eatery Blue/UI/PillButton/PillButtonView.swift:56]: Is this OK to be a constant?

**Review by tiffany-pann — CHANGES_REQUESTED** [review]
Amazing job with this feature Ant! 🚀 🎉 Just some small nitpicky stuff, feel free to rerequest me afterwards.

**Review by jjennifergu — APPROVED** [review]
Also skimmed through and looks good for MVP, but ideally we should not have any hard coded data, should be an action item to clean this up

**Review by tiffany-pann — APPROVED** [review]

File: Eatery Blue/UI/PillButton/PillButtonView.swift (hunk 12 above)
Comment (jjennifergu) [reply Eatery Blue/UI/PillButton/PillButtonView.swift:56]: Ideally it should resize wrt the screen but it should be ok for the MVP

**Comment by vinnie4k** [comment]
Going to merge your PRs. @antoinztte, the conversations will still be open so that you can look back at them.

**Pull request merged**